Mathlib/Analysis/LocallyConvex/AbsConvexOpen.lean

```diff
@@ -0,0 +1,122 @@
+/-
+Copyright (c) 2022 Moritz Doll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Moritz Doll
+-/
+import Mathlib.Analysis.LocallyConvex.AbsConvex
+import Mathlib.Analysis.LocallyConvex.WithSeminorms
+import Mathlib.Analysis.Convex.Gauge
+
+/-!
+# Absolutely convex open sets
+
+A set `s` in a commutative monoid `E` equipped with a topology is said to be an absolutely convex
+open set if it is absolutely convex and open. When `E` is a topological additive group, the topology
+coincides with the topology induced by the family of seminorms arising as gauges of absolutely
+convex open neighborhoods of zero.
+
+## Main definitions
+
+* `AbsConvexOpenSets`: sets which are absolutely convex and open
+* `gaugeSeminormFamily`: the seminorm family induced by all open absolutely convex neighborhoods
+  of zero.
+
+## Main statements
+
+* `with_gaugeSeminormFamily`: the topology of a locally convex space is induced by the family
+  `gaugeSeminormFamily`.
+
+-/
+
+open NormedField Set
+
+open NNReal Pointwise Topology
+
+variable {𝕜 E : Type*}
+
+section AbsolutelyConvexSets
+
+variable [TopologicalSpace E] [AddCommMonoid E] [Zero E] [SeminormedRing 𝕜]
+variable [SMul 𝕜 E] [SMul ℝ E]
+variable (𝕜 E)
+
+/-- The type of absolutely convex open sets. -/
+def AbsConvexOpenSets :=
+  { s : Set E // (0 : E) ∈ s ∧ IsOpen s ∧ AbsConvex 𝕜 s }
+
+noncomputable instance AbsConvexOpenSets.instCoeOut : CoeOut (AbsConvexOpenSets 𝕜 E) (Set E) :=
```

Comment:
I'm sorry, but I disagree with @j-loreaux here. The change from `CoeTC` to `CoeOut` needs to be in a separate PR.

Reply:
Fair enough.

Reply:
I've opened a separate PR for that: #22010 22010